filters/openpolicyagent/tracing_test.go

```diff
@@ -24,13 +25,13 @@ func (t *MockTransport) RoundTrip(*http.Request) (*http.Response, error) {
 }
 
 func TestTracingFactory(t *testing.T) {
-	tracer := &tracingtest.Tracer{}
+	tracer := tracingtest.NewTracer()
 
 	testCases := []struct {
 		name       string
 		req        *http.Request
 		tracer     opentracing.Tracer
-		parentSpan opentracing.Span
+		parentSpan string
```

Comment:
This change is because closing parent span created during test case setup after tracer.Reset() is wrong and does not work properly with MockTracer.